x-pack/plugins/observability_solution/observability_ai_assistant/README.md

```diff
@@ -50,24 +42,200 @@ The user can then also continue the conversation in a flyout by clicking "Start
+- `data`: this is an array of key/value pairs along with a description. This is not sent over to the LLM immediately, but the LLM is made aware of its existence. It can then pull in this data via a function request. This is done to keep the token cost down initially, but still keep the option open for the LLM to use it if it deems it necessary.
+- `actions`: actions are functions that are executed _client-side_ instead of server-side. That means that these functions can easily interact with the page, e.g. via React state updates or other API calls. For instance, you can use this to allow the Assistant to navigate to another page or fill in a form.
+
+From the start contract, you can use `service.setScreenContext`:
+
+```ts
+useEffect(() => {
+  return observabilityAIAssistant?.service.setScreenContext({
+    description: 'Description of the page',
+  });
+}, [observabilityAIAssistant?.service]);
+```
+
+`setScreenContext` returns a function that removes the screen context when called. **It's important that this function is returned to the `useEffect` hook,** to make sure it's called when its dependencies change, or when the component unmounts. If this doesn't happen, you'll duplicate context.
+
+Here's how to add data:
+
+```ts
+useEffect(() => {
+  return observabilityAIAssistant?.service.setScreenContext({
+    data: [
+      {
+        name: 'my_data',
+        description: 'All of my data',
+        value: allMyData,
+      },
+    ],
+  });
+}, [observabilityAIAssistant?.service, allMyData]);
+```
+
+`allMyData` here is sent over the wire to the LLM, so it needs to be serializable (e.g., you cannot include functions, but you can include structured data).
+
+Finally, for actions:
+
+```ts
+import { createScreenContext } from '@kbn/observability-ai-assistant-plugin/public';
+useEffect(() => {
+  return observabilityAIAssistant?.service.setScreenContext({
+    actions: createScreenContextAction(
+      {
+        name: 'navigate_to_home',
+        description: 'Navigate to the home page',
+        parameters: {
+          type: 'object',
+          properties: {
+            myProperty: 'myValue',
+          },
+        },
+      },
+      async ({ arguments: { myProperty } }) => {
+        return observabilityAIAssistant?.service.navigate(() => {
+          moveToUrl();
+        });
+      }
+    ),
+  });
+}, [observabilityAIAssistant?.service, moveToUrl]);
+```
+
+`navigate` is a utility function that waits until the network is silent for at least a second, and then sends the screen context from the new page back to the Assistant. `createScreenContext` is a utility
+
+You can also update some state and directly return data to the Assistant:
+
+```ts
+import { createScreenContext } from '@kbn/observability-ai-assistant-plugin/public';
+useEffect(() => {
+  return observabilityAIAssistant?.service.setScreenContext({
+    actions: createScreenContextAction(
```

Comment:
From this example it's not clear to me what `createScreenContextAction` does or how the action is invoked. An example where `myProperty` was actually used and made sense would help

Reply:
yeah I forgot to finish the `createScreenContext` explanation above. will add!